src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -220,15 +220,29 @@ trait MarathonConf
     noshort = true
   )
 
+  lazy val mesosAuthentication = toggle("mesos_authentication",
+    default = Some(false),
+    noshort = true,
+    descrYes = "Will enforce framework authentication while registering with Mesos with principal and optional secret.",
+    descrNo = "(Default) will not use framework authentication while registering with Mesos.",
+    prefix = "disable_"
+  )
+  dependsOnAll(mesosAuthentication, List(mesosAuthenticationPrincipal))
+
   lazy val mesosAuthenticationPrincipal = opt[String]("mesos_authentication_principal",
     descr = "Mesos Authentication Principal.",
     noshort = true
   )
 
   lazy val mesosAuthenticationSecretFile = opt[String]("mesos_authentication_secret_file",
+    descr = "Mesos Authentication Secret written to a file.",
```

Comment:
s/Mesos Authentication Secret written to a file./Path to a file containing the Mesos Authentication Secret/ ?